nmrfx-processor/src/main/java/org/nmrfx/processor/optimization/Fitter2.java

```diff
@@ -0,0 +1,320 @@
+package org.nmrfx.processor.optimization;
+
+import org.apache.commons.math3.analysis.MultivariateFunction;
+import org.apache.commons.math3.exception.DimensionMismatchException;
+import org.apache.commons.math3.exception.NotPositiveException;
+import org.apache.commons.math3.exception.NotStrictlyPositiveException;
+import org.apache.commons.math3.exception.TooManyEvaluationsException;
+import org.apache.commons.math3.optim.*;
+import org.apache.commons.math3.optim.nonlinear.scalar.GoalType;
+import org.apache.commons.math3.optim.nonlinear.scalar.ObjectiveFunction;
+import org.apache.commons.math3.optim.nonlinear.scalar.noderiv.CMAESOptimizer;
+import org.apache.commons.math3.random.RandomGenerator;
+import org.apache.commons.math3.random.SynchronizedRandomGenerator;
+import org.apache.commons.math3.random.Well19937c;
+import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
+import org.apache.commons.math3.util.FastMath;
+import org.codehaus.commons.compiler.CompileException;
+import org.codehaus.janino.ExpressionEvaluator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.function.BiFunction;
+import java.util.stream.IntStream;
+
+// Fitter2 allows multiple yValues per index (compared to Fitter, which allows only 1).
+// we may refactor so Fitter2 extends, or replaces Fitter
+
+public class Fitter2 {
+
+    private static final Logger log = LoggerFactory.getLogger(Fitter2.class);
+    static RandomGenerator random = new SynchronizedRandomGenerator(new Well19937c());
+
+    boolean reportFitness = false;
+    int reportAt = 10;
+    double[][] parValues;
+    double[][] xValues;
+    double[][] yValues;
+    double[][] errValues;
+
+    double[] lowerBounds;
+    double[] upperBounds;
+    double[] start;
+    double inputSigma;
+    BiFunction<double[], double[], double[]> function;
+    BiFunction<double[], double[][], Double> valuesFunction = null;
+    ExpressionEvaluator expressionEvaluator = null;
+    private Fitter2() {
+
+    }
+
+    public static Fitter getFitter(BiFunction<double[], double[], Double> function) {
+        Fitter fitter = new Fitter();
+        fitter.function = function;
+        return fitter;
+    }
```

Comment:
Fitter2 supports fitting multiple y values as is done for the reaction fitting (reactant and product) and duplicates some code of Fitter.  We should, if there is little performance hit, refactor it so it can effectively replace Fitter